src/parser/toplevel.lisp

```diff
@@ -604,6 +629,20 @@ consume all attributes"))))
 
     package))
 
+(defun parse-lisp-toplevel-form (form program)
+  "Parse lisp forms that are to be literally included in compiler output.
+
+If the outermost form matches (eval-when (compile-toplevel) ..), evaluate the enclosed forms."
+  (loop :for form :in (cst:raw (cst:rest form))
+        :when (and (eql (car form) 'eval-when)
+                   (some (lambda (k)
```

Comment:
Optimization Nit. Might be worth factoring this lambda out since it does not closer over any outer binding environment. Either that or `#.(lambda (k) (string-equal k 'compile-toplevel)`.

Reply:
I moved this to a helper eval-toplevel-p function and poked around at the disassembly -- it looks like it's not allocating, and SBCL didn't like #.:

```
toplevel.lisp:719:8:
  error: Objects of type COMPILED-FUNCTION can't be dumped into fasl files.
  note: The first argument never returns a value.
```